frontend/src/pages/RecipeSearch/RecipeSearch.jsx

```diff
@@ -73,38 +100,258 @@ export default function RecipeSearch() {
+                                    {" "}
+                                    An error has occured, please try searching
+                                    again.{" "}
+                                </h3>
+                            ) : (
+                                <h3>
+                                    {" "}
+                                    Search for Ingredients to show Recipe
+                                    Results.{" "}
+                                </h3>
+                            )}
+                        </ul>
+                    </div>
+                </section>
+            </section>
+        </StyledRecipeSearch>
+    );
+}
 
-            <div>
-                {ingredients}
-            </div>
+/// APPLY FILTER *****************************************
+function applyFilter(recipeListArr, recipeFilters, categoryFilters) {
+    let filteredList = recipeListArr;
 
-            <div className='filter'>
-                <span>Filter</span>
-                <form action="">
-                    <label htmlFor=""></label>
-                </form>
-            </div>
+    //no filters at all or recipe filters array is empty
+    if (
+        (!recipeFilters && !categoryFilters) ||
+        (recipeFilters.length <= 0 && categoryFilters.length <= 0)
+    ) {
+        return recipeListArr;
+    }
 
-            <div className='searchResults'>
-                {searchSpinner ? <SearchSpinner /> :  recipeList.length > 0 ? recipeList.map(recipe => (
+    //process recipes through diets filters
+    if (recipeFilters.length > 0) {
+        filteredList = recipeListArr.filter((recipe) => {
+            let recipeCategories = recipe.diets;
 
-                    <RecipeCard key={recipe.id} recipe={recipe} />
-                )) :
-                    error ? <h3> An error has occured, please try searching again. </h3>
-                        : <h3> Search for Ingredients to show Recipe Results. </h3>
+            // loop through recipe filters
+            for (let i = 0; i < recipeFilters.length; i++) {
+                let currentFilter = recipeFilters[i];
+                // recipe fails to include all filters
+                if (!recipeCategories.includes(currentFilter)) {
+                    return false;
                 }
-                
-            </div>
-                
-        </StyledRecipeSearch>
-    )
+            }
+            return true;
+        });
+    }
+
+    //process recipes through categories by checking recipe specific boolean properties
+    // and update filteredList
+    if (categoryFilters.length > 0) {
+        filteredList = filteredList.filter((recipe) => {
+            for (let i = 0; i < categoryFilters.length; i++) {
+                if (!recipe[categoryFilters[i]]) {
+                    return false;
+                }
+            }
+            return true;
+        });
+    }
+
+    return filteredList;
 }
 
+//=====================================
```

Comment:
It's typically best practice to pull something like this out as a separate component in a separate file, and then import it into the "page" that it is displayed on.  
There's a components folder in the frontend project, I'd like to see you create a separate FilterList component and put it in that folder, and then import it into this RecipeSearch page, passing the props as needed.  I think this would also involve pulling out the specific styles that are needed for the Filter component and putting those in a FilterList.styles.js file as well.

Reply:
ok I moved the filterList component out of the recipeSerach.jsx

Reply:
just noticed there are conflicts il see if i can resolve them

Reply:
ok everything seems to be working now